ts/sagas/pinlogin.ts

```diff
@@ -30,13 +44,30 @@ function* pinValidateSaga(action: PinValidateRequest): Iterator<Effect> {
   try {
     const userPin = action.payload;
     const basePin: string = yield call(getPin);
+
     if (basePin === userPin) {
-      // Navigate to the MainNavigator
-      const navigateToPinValidNavigatorAction = NavigationActions.navigate({
-        routeName: ROUTES.MAIN,
-        key: undefined
-      });
-      yield put(navigateToPinValidNavigatorAction);
+      const deeplink: NavigationNavigateActionPayload | null = yield select(
+        deeplinkSelector
+      );
+      // tslint:disable-next-line:readonly-array array-type
+      const actionsToDispatch: Array<PutEffect<Action>> = [];
+
+      // If a deeplink has been setted, navigate to deeplink, otherwise to the MainNavigator
+      if (deeplink) {
```

Comment:
Relying on a mutable array seems unneeded, do the following instead (that is also easier to read):

```ts
if (deeplink) {
  // ....
  yield put(navigateToDeeplink(deeplink));
  // ....
  yield put(clearDeeplink());
} else {
  // ....
  yield put(
            NavigationActions.navigate({
              routeName: ROUTES.MAIN,
              key: undefined
            })
    );
}
```